src/templates.hpp

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2009-2010 Satoshi Nakamoto
+// Copyright (c) 2009-2014 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_TEMPLATES_H
+#define BITCOIN_TEMPLATES_H
+
+#include <assert.h>
+#include <cstddef>
+
+template<typename T>
+class Container
```

Comment:
What is the purpose of this `Container`? 
It seems to be managing the lifetime of a pointer,  is a boost managed pointer not suitable?